### ISSUE TITLE
Document "unistore"

### ISSUE THREAD

**Issue by kolbe** [pingcap/docs#5159]
## Change Request

Please answer the following questions before submitting your issue. Thanks!

1. Describe what you find is inappropriate or missing in the existing docs.

https://github.com/pingcap/tidb/pull/17156 added "unistore" as the default standalone storage engine instead of mocktikv, but somehow it's not mentioned at all in our documentation!

2. Describe your suggestion or addition.

We should describe what "unistore" is, how it's different than "mocktikv", etc.

3. Provide some reference materials (documents, websites, etc) if you could.

**Comment by TomShawn** [comment]
@coocood @lysu PTAL whether this can be documented, thanks!

**Comment by TomShawn** [comment]
docs-cn PR: https://github.com/pingcap/docs-cn/pull/5874

**Comment by TomShawn** [comment]
docs PR: https://github.com/pingcap/docs/pull/5160
@kolbe PTAL whether your issue has been addressed or not.

**Comment by kolbe** [comment]
@TomShawn not quite. It seems that "mocktikv" is still an option, alongside "unistore" and "tikv":

```
  -store string
        registered store name, [tikv, mocktikv, unistore] (default "unistore")
```

The documentation should reflect that all 3 options are available.

**Comment by TomShawn** [comment]
> @TomShawn not quite. It seems that "mocktikv" is still an option, alongside "unistore" and "tikv":
> 
> ```
>   -store string
>         registered store name, [tikv, mocktikv, unistore] (default "unistore")
> ```
> 
> The documentation should reflect that all 3 options are available.

@lysu @coocood @tangenta PTAL again, thanks!

**Comment by coocood** [comment]
@kolbe
‘mocktikv' is not maintained anymore, it doesn't support clustered index.
I think we don't need to mention it in the doc.

**Comment by kolbe** [comment]
It's mentioned in the help output of `tidb-server --help`! We shouldn't hide things in the documentation just because they're no longer maintained or useful, if they're still included in the product. In fact, just the opposite is true — if something is obviously a valid option, but we don't think anyone should use it, we must explain in the documentation what it is and why it should be avoided!

**Comment by TomShawn** [comment]
> It's mentioned in the help output of `tidb-server --help`! We shouldn't hide things in the documentation just because they're no longer maintained or useful, if they're still included in the product. In fact, just the opposite is true — if something is obviously a valid option, but we don't think anyone should use it, we must explain in the documentation what it is and why it should be avoided!

@coocood What do you think?

**Comment by coocood** [comment]
I think we need to remove the `mocktikv` option in `tidb-server`

**Comment by TomShawn** [comment]
> I think we need to remove the `mocktikv` option in `tidb-server`

@kolbe Then we do not need to mention `mocktikv` in the docs. Is that OK?
